File: Part 2 - Regression/Section 9 - Random Forest Regression/R/random_forest_regression.R
# RANDOM FOREST REgression

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


# Fitting Decision Tree REgression
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 500)
#ntree sets number of tree

####PREDICTING the result
y_pred = predict(regressor, data.frame(Level = 6.5))

#Visualizing the result
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot()+
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue')+
  ggtitle('Truth or Bluff(Random ForestRegression)')+
  xlab('Position')+
  ylab('Salary')

File: Part 2 - Regression/Section 9 - Random Forest Regression/Python/random_forest_regression.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 29 19:40:55 2020

@author: Dell
"""
                    # Random Forest Regression
                    
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

dts = pd.read_csv('Position_Salaries.csv')
x = dts.iloc[:, 1:-1].values
y = dts.iloc[:, -1].values


'''Training the Random Forest Regression'''
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=30, random_state=0)
regressor.fit(x, y)

## Predicting a new result
regressor.predict([[6.5]])
    

##Visualising the Decision Tree Regression results (higher resolution)
x_grid = np.arange(min(x), max(x), 0.1)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color='red')
plt.plot(x_grid, regressor.predict(x_grid), color='blue')
plt.title('Truth or Bluff (Random Forest Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()


